build/integration.sh

```diff
@@ -56,6 +56,22 @@ while [ "$(curl -Gs http://localhost:8080/healthz)" != "ok" ]; do
   sleep 1
 done
 
+if [[ "${DOCKER_IN_DOCKER_ENABLED:-}" == "true" ]]; then
+  # see https://github.com/moby/moby/blob/master/hack/dind
+  # cgroup v2: enable nesting
+  if [ -f /sys/fs/cgroup/cgroup.controllers ]; then
+    echo ">> configuring cgroupsv2 for docker in docker..."
+    # move the processes from the root group to the /init group,
+    # otherwise writing subtree_control fails with EBUSY.
+    # An error during moving non-existent process (i.e., "cat") is ignored.
+    mkdir -p /sys/fs/cgroup/init
+    xargs -rn1 < /sys/fs/cgroup/cgroup.procs > /sys/fs/cgroup/init/cgroup.procs || :
```

Comment:
What does this xargs block do?

I can see that `-r` is do not run the command if there is no stdin, and `-n1 is max args 1?` but I don't see a command that its executing.

Reply:
I copied this over from https://github.com/moby/moby/blob/master/hack/dind :)

So the idea here, is that for docker in docker for cgroupv2 we need to be move all the processes in a child cgroup or else we will hit the runc/cgroupv2 limitation mentioned here - https://github.com/containerd/containerd/issues/6659#issuecomment-1064754423

The goal here is to first create a new cgroup (`mkdir -p /sys/fs/cgroup/init`). Then we want to copy all of pids one by one from `/sys/fs/cgroup/cgroup.procs` to the new child cgroup `/sys/fs/cgroup/init/cgroup.procs` (the xargs `xargs -rn1` ensures it's split one by one). 

There is no command being passed to xargs, so as a result it will simply echo each pid (see https://unix.stackexchange.com/questions/110134/what-does-xargs-do-if-its-used-without-any-parameter)

Reply:
Neat, so xargs allows you to copy the file but throw an error if the source doesn't have any lines in it.

The new comment is helpful thanks!